src/ruleset/v3/functions/requiredOperationChannelUnambiguity.ts

```diff
@@ -0,0 +1,42 @@
+import { createRulesetFunction } from '@stoplight/spectral-core';
+import type { IFunctionResult } from '@stoplight/spectral-core';
+import { SchemaDefinition } from '@stoplight/spectral-core/dist/ruleset/function';
+
+const referenceSchema: SchemaDefinition = {
+  type: 'object',
+  properties: {
+    $ref: {
+      type: 'string',
+      format: 'uri-reference'
+    },
+  },
+};
+
+export const requiredOperationChannelUnambiguity = createRulesetFunction<{ channel?: {'$ref': string}; messages?: [{'$ref': string}] }, null>(
+  {
+    input: {
+      type: 'object',
+      properties: {
+        channel: referenceSchema,
+        messages: {
+          type: 'array',
+          items: referenceSchema,
+        },
+      },
+    },
+    options: null,
+  },
+  (targetVal, _, ctx) => {
+    const results: IFunctionResult[] = [];
+    const channelPointer = targetVal.channel?.$ref as string; // required
+
+    if (channelPointer.includes('#/components/channels')) {
```

Comment:
I think this is incomplete. It should be something like:

```suggestion
    if (!channelPointer.includes('#/channels/')) {
```

Everything that's not pointing to root channels is not allowed. Right?

Reply:
Agree, `if (!channelPointer.includes('#/channels/')) {` is more accurate as you could have it pointing to different locations then `components`, which is still incorrect